schema-registry/client/src/main/java/com/hortonworks/registries/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -251,18 +276,34 @@ public SchemaVersionInfo retrieveSchemaVersion(SchemaIdVersion key) throws Schem
 
         SchemaMetadataCache.SchemaMetadataFetcher schemaMetadataFetcher = createSchemaMetadataFetcher();
         schemaMetadataCache = new SchemaMetadataCache(((Number) configuration.getValue(Configuration.SCHEMA_METADATA_CACHE_SIZE
-                                                                                               .name())).longValue(),
-                                                      ((Number) configuration.getValue(Configuration.SCHEMA_METADATA_CACHE_EXPIRY_INTERVAL_SECS
-                                                                                               .name())).longValue(),
-                                                      schemaMetadataFetcher);
+                .name())).longValue(),
+                ((Number) configuration.getValue(Configuration.SCHEMA_METADATA_CACHE_EXPIRY_INTERVAL_SECS
+                        .name())).longValue(),
+                schemaMetadataFetcher);
 
         schemaTextCache = CacheBuilder.newBuilder()
-                                      .maximumSize(((Number) configuration.getValue(Configuration.SCHEMA_TEXT_CACHE_SIZE
-                                                                                            .name())).longValue())
-                                      .expireAfterAccess(((Number) configuration.getValue(Configuration.SCHEMA_TEXT_CACHE_EXPIRY_INTERVAL_SECS
-                                                                                                  .name())).longValue(),
-                                                         TimeUnit.SECONDS)
-                                      .build();
+                .maximumSize(((Number) configuration.getValue(Configuration.SCHEMA_TEXT_CACHE_SIZE
+                        .name())).longValue())
+                .expireAfterAccess(((Number) configuration.getValue(Configuration.SCHEMA_TEXT_CACHE_EXPIRY_INTERVAL_SECS
+                                .name())).longValue(),
+                        TimeUnit.SECONDS)
+                .build();
+    }
+
+    private void initializeRetryPolicy(String retryPolicyClass, Map<String, Object> retryPolicyProps) {
+        ClassLoader classLoader = this.getClass().getClassLoader();
+        Class<? extends RetryPolicy> clazz = null;
+        try {
+            clazz = (Class<? extends RetryPolicy>) Class.forName(retryPolicyClass, true, classLoader);
```

Comment:
I am not quite sure. At the time when I plugged this in, it made sense, now I am not quite sure. Do you guys think this could be a real requirement? @guruchai

Reply:
Not necessary. It picks current class's classloader anyway.